apps/mitt-konto/src/Components/Paywall.purs

```diff
@@ -0,0 +1,217 @@
+module MittKonto.Components.Paywall where
+
+import Prelude
+
+import Data.Array ((:), filter, findIndex, head, modifyAt, null)
+import Data.Int (fromString)
+import Data.Maybe (Maybe(..), fromMaybe, maybe)
+import Effect (Effect)
+import Effect.Aff as Aff
+import Effect.Class (liftEffect)
+import Effect.Exception (error)
+import Effect.Uncurried (mkEffectFn1)
+import Data.JSDate (JSDate, now, toUTCString)
+import KSF.Api.Entitlements (PaywallOpening(..))
+import KSF.Sentry as Sentry
+import KSF.Spinner (loadingSpinner)
+import KSF.User as User
+import React.Basic (JSX)
+import React.Basic.DOM as DOM
+import React.Basic.DOM.Events (targetValue)
+import React.Basic.Events (EventHandler, handler, handler_)
+import React.Basic.Hooks (Component, component, useEffect, useState, useState', (/\))
+import React.Basic.Hooks as React
+import Routing.PushState (PushStateInterface)
+
+type Props = {}
+type Product = { name :: String, label :: String, selected :: Boolean }
+type Products = Array Product
+
+initialProducts :: Products
+initialProducts =
+  [ { name: "hbl-365", label: "HBL 365", selected: false }
+  , { name: "hbl-epaper", label: "HBL e-tidning", selected: false }
+  , { name: "hbl-web", label: "HBL webbplats", selected: false }
+  , { name: "junior-epaper", label: "HBL Junior e-tidning", selected: false }
+  , { name: "on-365", label: "Östra Nyland 365", selected: false }
+  , { name: "on-epaper", label: "Östra Nyland e-tidning", selected: false }
+  , { name: "on-web", label: "Östra Nyland webbplats", selected: false }
+  , { name: "vn-365", label: "Västra Nyland 365", selected: false }
+  , { name: "vn-epaper", label: "Västra Nyland e-tidning", selected: false }
+  , { name: "vn-web", label: "Västra Nyland webbplats", selected: false }
+  ]
+
+paywall :: PushStateInterface -> Sentry.Logger -> Component Props
+paywall _router _logger = do
+  currentDate <- now
+  component "Paywall" \ {} -> React.do
+    days     /\ setDays     <- useState' 0
+    hours    /\ setHours    <- useState' 0
+    minutes  /\ setMinutes  <- useState' 0
+    products /\ setProducts <- useState initialProducts
+    openings /\ setOpenings <- useState' Nothing
+    era      /\ setEra      <- useState 0
+    useEffect era $ do
+      fiber <- Aff.launchAff $
+        liftEffect <<< setOpenings <<< Just =<< User.getPaywallOpenings
+      pure $ Aff.launchAff_ $ Aff.killFiber (error "component closed") fiber
+    let
+      selection = map (\p -> p.name) $ filter (\p -> p.selected) products
+      submitHandler :: EventHandler
+      submitHandler = handler_ $ Aff.launchAff_ $ do
+        User.openPaywall days hours minutes selection
+        liftEffect $ setEra (_ + 1)
+    pure $ DOM.div
```

Comment:
This could also use `DOM.form` at this level instead and use `onSubmit` to do what the `Skicka` button's `onClick` does. And use `type: "submit"` for that button, too.

Reply:
Using `DOM.input` with `type: "submit"` leads into page reload. The `type` attribute doesn't seem to be available for `DOM.button` on the other hand.

Reply:
Using a `DOM.form` element seems to lead into page reload after each click of a button.